src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java

```diff
@@ -0,0 +1,199 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.remoteconfig;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNull;
+
+import com.google.api.core.ApiFuture;
+import com.google.api.core.ApiFutureCallback;
+import com.google.api.core.ApiFutures;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.firebase.testing.IntegrationTestUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FirebaseRemoteConfigIT {
+
+  private static FirebaseRemoteConfig remoteConfig;
+
+  private static final long timestamp = System.currentTimeMillis();
+
+  private static final Map<String, Parameter> PARAMETERS = ImmutableMap.of(
+          "welcome_message_text", new Parameter()
+                  .setDefaultValue(ParameterValue
+                          .of(String.format("welcome to app %s", timestamp)))
+                  .setConditionalValues(ImmutableMap.<String, ParameterValue>of(
+                          "ios_en",
+                          ParameterValue.of(String.format("welcome to app en %s", timestamp))
+                  ))
+                  .setDescription("text for welcome message!"),
+          "header_text", new Parameter()
+                  .setDefaultValue(ParameterValue.inAppDefault()));
+
+  private static final Map<String, ParameterGroup> PARAMETER_GROUPS = ImmutableMap.of(
+          "new menu", new ParameterGroup()
+                  .setDescription(String.format("New Menu %s", timestamp))
+                  .setParameters(ImmutableMap.of(
+                          "pumpkin_spice_season", new Parameter()
+                                  .setDefaultValue(ParameterValue.of("true"))
+                                  .setDescription("Whether it's currently pumpkin spice season."))
+                  ));
+
+  private static final List<Condition> CONDITIONS = ImmutableList.of(
+          new Condition("ios_en", "device.os == 'ios' && device.country in ['us', 'uk']")
+                  .setTagColor(TagColor.INDIGO),
+          new Condition("android_en",
+                  "device.os == 'android' && device.country in ['us', 'uk']"));
+
+  private static final Version VERSION = Version
+          .withDescription(String.format("promo config %s", timestamp));
+
+  @BeforeClass
+  public static void setUpClass() {
+    remoteConfig = FirebaseRemoteConfig.getInstance(IntegrationTestUtils.ensureDefaultApp());
+  }
+
+  @Test
+  public void testTemplateOperations() throws FirebaseRemoteConfigException {
```

Comment:
Nit: Some of these can be split into separate test case with a bit of effort and extra RPCs (I'll leave it to you to decide whether that's worth it)

1. validate can be easily turned into a separate test case
2. getAtVersion and rollback can also be made into separate test cases if we are willing to run a few extra get and publish operations.

Reply:
Good point! As discussed offline let's address this in a future PR. Thanks!